Похоже, работает

Apoezon opened the pull request:
Основная логика программы завершена:
1) в командной строке задаются имена игроков и размер поля;
2) функция проверяет поле на наличие победителя (когда строка или столбец, или диагональ заполнена крестиком или ноликом) или ничью (если все поля заняты, но условие победы не достигнуто);
3) игрок ходит посредством ввода номера строки и стобца (осуществляется проверка правильности ввода)
4) манипуляции кореткой ввода и очистки терминала осуществляются с помощью выражений в кодировки ANSI (эти выражения не работают в терминале PyCharm)
Проверил, похоже, работает! :о)

Apoezon committed:
Основная логика программы завершена:
1) в командной строке задаются имена игроков и размер поля;
2) функция проверяет поле на наличие победителя (когда строка или столбец, или диагональ заполнена крестиком или ноликом) или ничью (если все поля заняты, но условие победы не достигнуто);
3) игрок ходит посредством ввода номера строки и стобца (осуществляется проверка правильности ввода)
4) манипуляции кореткой ввода и очистки терминала осуществляются с помощью выражений в кодировки ANSI (эти выражения не работают в терминале PyCharm)
Проверил, похоже, работает! :о)

EgorOb reviewed (approved):
Хорошо. Просто так много кода и исправлений, что сложно уследить. Необязательно почти для каждой функции свой файл создавать, достаточно под конкретный объём задач сделать файл, и сопровождающие функции. Тут достаточно 4 файла. Файл с запуском приложения, в котором main. Файл с кодом для отрисовки, файл с кодом логики и сопутствующих моделей. Файл с кодом общения пользователя с консолью и сопутствующих проверок. Для каждой функции свой файл это лишнее конечно, но не запрещается так делать. В практике полно случаев когда в одном файле собирается более 1000 строк кода, главное чтобы они были в одном векторе.

Apoezon committed:
Основная логика программы завершена:
1) в командной строке задаются имена игроков и размер поля;
2) функция проверяет поле на наличие победителя (когда строка или столбец, или диагональ заполнена крестиком или ноликом) или ничью (если все поля заняты, но условие победы не достигнуто);
3) игрок ходит посредством ввода номера строки и стобца (осуществляется проверка правильности ввода)
4) манипуляции кореткой ввода и очистки терминала осуществляются с помощью выражений в кодировки ANSI (эти выражения не работают в терминале PyCharm)
Проверил, похоже, работает! :о)
